### PR TITLE
Merge Create Service Wizard

### PR DESCRIPTION
**Pull request by morhag90**
Issue Number: #3043 

## PR Details

We want to provide new users with a smooth way to create a new service. This will be done via a wizard that will give the user choices how to generate his service :

Generation settings (graphql/rest api/ admin-ui)
Sample entities

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+};
+
+const CreateServiceWizard: React.FC<Props> = ({ moduleClass, match }) => {
+  const { workspace, project } = match.params;
+  const serviceSettingsFields: MutableRefObject<serviceSettings> = useRef(
+    serviceSettingsFieldsInitValues
+  );
+  const { trackEvent } = useTracking();
+
+  const [generalError, setGeneralError] = useState<Error | undefined>(
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx:61]: no need, removed

File: packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+  moduleClass: string;
+};
+
+type TData = {
+  createResourceWithEntities: models.Resource;
+  createResourceGenSettings: models.ResourceGenSettingsCreateInput;
+};
+
+export const serviceSettingsFieldsInitValues = {
+  generateAdminUI: true,
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx:48]: move it to constant

File: packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            data.createResourceWithEntities,
+          ]),
+        },
+      });
+    },
+  });
+
+  const errorMessage = formatError(error) || formatError(generalError);
+
+  const handleStartFromSample = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx:94]: Done

File: packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    trackEvent({
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx:95]: remove this functions and merge to one createResource

File: packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+  }, [history, data, project, workspace]);
+
+  const handleSubmitResource = (currentServiceSettings: serviceSettings) => {
+    serviceSettingsFields.current = currentServiceSettings;
+  };
+
+  const handleClick = () => {
+    if (!serviceSettingsFields) return;
+
+    if (serviceSettingsFields.current.resourceType === "sample") {
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx:160]: merge to one function

File: packages/amplication-client/src/Resource/create-resource/CreateServiceWizardForm.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import { serviceSettingsFieldsInitValues } from "./CreateServiceWizard";
+
+const CLASS_NAME = "create-service-wizard-form";
+
+type Props = {
+  handleSubmitResource: (serviceSettings: serviceSettings) => void;
+};
+
+export type serviceSettings = {
+  generateAdminUI: boolean;
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/Resource/create-resource/CreateServiceWizardForm.tsx:15]: move to constants

File: packages/amplication-client/src/models.ts [hunk 7]
```diff
@@ -1644,9 +1644,16 @@ export type ResourceCreateWithEntitiesFieldInput = {
 export type ResourceCreateWithEntitiesInput = {
   commitMessage: Scalars["String"];
   entities: Array<ResourceCreateWithEntitiesEntityInput>;
+  generationSettings: ResourceGenSettingsCreateInput;
   resource: ResourceCreateInput;
 };
 
+export type ResourceGenSettingsCreateInput = {
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/models.ts:1651]: this is a different type that came from the server @barshimi

File: packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+
+    createResourceWithEntities({
+      variables: { data: scratchResource },
+    }).catch(console.error);
+  }, [createResourceWithEntities, trackEvent, project, serviceSettingsFields]);
+
+  const history = useHistory();
+
+  useEffect(() => {
+    if (data) {
```
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/amplication-client/src/Resource/create-resource/CreateServiceWizard.tsx]: I checked resource creations with log inside useEffect and it look fine (write data only one time) @barshimi

**Review by barshimi — APPROVED** [review]

**Pull request merged**